kanidmd/src/lib/macros.rs

```diff
@@ -576,6 +577,33 @@ macro_rules! btreeset {
     });
 }
 
+#[allow(unused_macros)]
+#[macro_export]
+macro_rules! valueset {
+    () => (
+        compile_error!("ValueSet needs at least 1 element")
+    );
+    ($e:expr) => ({
+        use crate::valueset::ValueSet;
+        let mut x: ValueSet = ValueSet::new();
+        assert!(x.insert($e));
+        x
+    });
+    ($e:expr,) => ({
+        use crate::valueset::ValueSet;
+        let mut x: ValueSet = ValueSet::new();
+        assert!(x.insert($e));
+        x
```

Comment:
No need to repeat
```suggestion
        valueset!($e)
```

Reply:
I think there are issues with recursive definitions like this.

Reply:
In general or in this specific scenario? Because I know they're used frequently like here https://docs.rs/tracing/0.1.26/src/tracing/macros.rs.html#586-792

Reply:
I think I've had issues with them before so maybe I should just try it :)

Reply:
Well damn. It works. Doing this :)